datasource/src/test/scala/quasar/plugin/sqlserver/datasource/SQLServerDatasourceSpec.scala

```diff
@@ -242,4 +259,179 @@ object SQLServerDatasourceSpec extends TestHarness with Logging {
       }
     }
   }
+
+  "seek data" >> {
+    import DataPathSegment._
+
+    "errors when path is incorrect" >> {
+      val key = ∃(InternalKey.Actual.string("1"))
+      val twoFields = Offset.Internal(NonEmptyList.of(Field("foo"), Field("bar")), key)
+      val index = Offset.Internal(NonEmptyList.one(Index(0)), key)
+      val all = Offset.Internal(NonEmptyList.one(AllFields), key)
+      val allIndices = Offset.Internal(NonEmptyList.one(AllIndices), key)
+      val external = Offset.External(ExternalOffsetKey(Array(0x01, 0x01)))
+
+      harnessed() use { case (xa, ds, path, name) =>
+        for {
+          _ <- (fr"CREATE TABLE" ++ Fragment.const0(name) ++ fr0" (col INTEGER)").update.run.transact(xa)
+          two <- seekRValues(ds, path, twoFields).attempt
+          ind <- seekRValues(ds, path, index).attempt
+          allF <- seekRValues(ds, path, all).attempt
+          allI <- seekRValues(ds, path, allIndices).attempt
+          ext <- seekRValues(ds, path, external).attempt
+        } yield {
+          two must beLeft
+          ind must beLeft
+          allF must beLeft
+          allI must beLeft
+          ext must beLeft
+        }
+      }
+    }
+
+    "string offset" >> {
+      harnessed() use { case (xa, ds, path, name) =>
+        val setup =
+          (fr"CREATE TABLE" ++ Fragment.const0(name) ++ fr0" (id INTEGER, offset VARCHAR(32))").update.run >>
+            (fr"INSERT INTO" ++ Fragment.const0(name) ++ fr0" VALUES (0, 'foo')").update.run >>
+            (fr"INSERT INTO" ++ Fragment.const0(name) ++ fr0" VALUES (1, 'bar')").update.run >>
+            (fr"INSERT INTO" ++ Fragment.const0(name) ++ fr0" VALUES (2, 'baz')").update.run
+
+        val offset =
+          Offset.Internal(NonEmptyList.one(Field("OFFSET")), ∃(InternalKey.Actual.string("baz")))
+
+        val expected = List(
+          obj("id" -> rLong(2L), "offset" -> rString("baz")),
+          obj("id" -> rLong(0L), "offset" -> rString("foo")))
```

Comment:
Is `foo` expected in the output here intentionally? I thought it would be just `baz`?

Reply:
Yes, it's correct AFAICT, `"foo" > "baz"`